ydb/library/yql/providers/generic/provider/yql_generic_load_meta.cpp

```diff
@@ -226,14 +167,64 @@ namespace NYql {
-                                              const std::string_view& table, TExprContext& ctx, TVector<TString>& columnOrder) try {
+        TIssues HandleDescribeTableResponse(
+            const TIntrusivePtr<TExprNode>& read,
+            TExprContext& ctx,
+            TNodeOnNodeOwnedMap& replaces
+        ) {
+            const TGenRead genRead(read);
+            const auto clusterName = genRead.DataSource().Cluster().StringValue();
+            const auto& keyArg = TExprBase(genRead.FreeArgs().Get(2).Ref().HeadPtr()).Cast<TCoKey>().Ref().Head();
+            const auto tableName = TString(keyArg.Tail().Head().Content());
+
+            const auto it = Results_.find(TGenericState::TTableAddress(clusterName, tableName));
+
+            if (it == Results_.cend()) {
+                TIssues issues;
+                issues.AddIssue(TIssue(ctx.GetPosition(genRead.Pos()), TStringBuilder()
+                                << "Not found result for " << clusterName << '.' << tableName));
+                return issues;
+            }
+
+            const auto& response = it->second->Response;
+
+            if (!NConnector::IsSuccess(*response)) {
+                return NConnector::ErrorToIssues( 
+                    response->error(),
+                    TStringBuilder() << "Loading metadata for table: " << clusterName << '.' << tableName
+                );
+            }
+
+            TGenericState::TTableMeta tableMeta;
+            tableMeta.Schema = response->schema();
+            tableMeta.DataSourceInstance = it->second->DataSourceInstance;
+
+            auto issues = ParseTableMeta(ctx, ctx.GetPosition(read->Pos()), clusterName, tableName, tableMeta);
+            if (issues) {
+                return issues;
+            }
+
+            if (const auto ins = replaces.emplace(genRead.Raw(), TExprNode::TPtr()); ins.second) {
+                ins.first->second = MakeTableMetaNode(ctx, genRead, tableName);
+            }
+
+            State_->AddTable(clusterName, tableName, std::move(tableMeta));
+            return TIssues{};
+        }
+
+        TIssues ParseTableMeta(
+            TExprContext& ctx,
+            const TPosition& pos,
+            const std::string_view& cluster,
+            const std::string_view& table,
+            TGenericState::TTableMeta& tableMeta
+        ) try {
             TVector<const TItemExprType*> items;
 
-            auto columns = schema.columns();
+            auto columns = tableMeta.Schema.columns();
             if (columns.empty()) {
-                ctx.AddError(TIssue({}, TStringBuilder() << "Table " << cluster << '.' << table << " doesn't exist."));
-                return nullptr;
+                TIssues issues;
+                issues.AddIssue(TIssue(pos, TStringBuilder() << "Table " << cluster << '.' << table << " doesn't exist."));
             }
 
             for (auto i = 0; i < columns.size(); i++) {
```

Comment:
Nit: can be replaced with `for(const auto& column: columns) { ` and `columns.Get(i)` with `column`

Reply:
fixed